great_expectations/core/id_dict.py

```diff
@@ -4,16 +4,21 @@
 import json
 from typing import Any, Set, TypeVar, Union
 
+from typing_extensions import Never
+
 from great_expectations.compatibility.typing_extensions import override
 from great_expectations.util import convert_to_json_serializable  # noqa: TID251 # FIXME CoP
 
 T = TypeVar("T")
 
 
+IDDictID = Union[str, tuple[Never, ...]]
```

Comment:
i'm having trouble when using this type with pydantic. did you mean `IDDictID = Union[str, tuple[()]]`? Or if that isn't what you meant, what does `Never` stand for here?

Reply:
@joshua-stauffer I was able to get the static analysis to work with `Union[str, tuple[()]]` if I updated `tuple()` calls to `()` when instantiating `MetricConfigurationID`. If the tests pass I will merge it like this. I think `tuple[Never, ...]` means something like "a tuple that will never have anything in it" which is semantically the same. Both `tuple()` and `()` create an empty tuple, so if this works better for your PR I will merge it this way.